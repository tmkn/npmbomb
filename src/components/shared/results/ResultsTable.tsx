/** @jsx jsx */
import { jsx, css } from "@emotion/react";

import { serifFont, textColor, mq } from "../../../css";
import { PropsWithChildren } from "react";

const numberStyle = css({
    [mq[0]]: {
        color: `${textColor}`,
        fontFamily: `"${serifFont}"`
    }
});

export const Num: React.FC<PropsWithChildren> = ({ children }) => (
    <span css={numberStyle}>{children}</span>
);

interface IResultsTableProps {
    columns: number;
}

export const ResultsTable: React.FC<PropsWithChildren<IResultsTableProps>> = ({
    children,
    columns
}) => {
    const resultsTable = css({
        [mq[0]]: {
            display: "grid",
            gridTemplateColumns: `${new Array(columns)
                .fill("")
                .map(n => "1fr")
                .join(" ")}`,
            columnGap: "1rem",
            rowGap: "1rem",
            color: `${textColor}`
        }
    });

    return <div css={resultsTable}>{children}</div>;
};
