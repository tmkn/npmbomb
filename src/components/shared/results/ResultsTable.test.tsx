import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import { ResultsTable, Num } from "./ResultsTable";

describe("<ResultsTable />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(
            <ResultsTable columns={2}>
                <div>1</div>
                <div>
                    <Num>2</Num>
                </div>
                <div>3</div>
                <div>4</div>
            </ResultsTable>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
