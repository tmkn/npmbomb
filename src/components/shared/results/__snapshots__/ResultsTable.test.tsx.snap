// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsTable /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-ziu505-ResultsTable"
  >
    <div>
      1
    </div>
    <div>
      <span
        class="css-1lpv6qu-Num"
      >
        2
      </span>
    </div>
    <div>
      3
    </div>
    <div>
      4
    </div>
  </div>
</DocumentFragment>
`;
